src/components/notes/editor-view.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useCallback, useRef, useEffect, useMemo } from 'react'
+
+import { Editor, WithEditorActions, EditorContext } from '@atlaskit/editor-core'
+import { MainTitle } from 'cozy-ui/transpiled/react/Text'
+import Textarea from 'cozy-ui/transpiled/react/Textarea'
+import { useI18n } from 'cozy-ui/transpiled/react/I18n'
+import useEventListener from 'cozy-ui/transpiled/react/hooks/useEventListener'
+
+import editorConfig, {
+  imageUploadHandler
+} from 'components/notes/editor_config'
+import HeaderMenu from 'components/header_menu'
+import styles from 'components/notes/editor-view.styl'
+import { ImageUploadProvider } from '@atlaskit/editor-common/dist/types/provider-factory'
+
+function updateTextareaHeight(target) {
+  if (target) target.style.height = `${target.scrollHeight}px`
+}
+
+const TestPlugin = <p>Plugin</p>
+const nullCallback = () => {}
+
+function EditorView(props) {
+  const {
+    defaultValue,
+    onTitleChange,
+    onTitleBlur,
+    defaultTitle,
+    title,
+    collabProvider,
+    leftComponent,
+    rightComponent,
+    onContentChange,
+    readOnly,
+    bannerRef,
+    serviceClient,
+    noteId
+  } = props
+  const { t } = useI18n()
+  const titleEl = useRef(null)
+
+  const onTitleEvent = useCallback(
+    e => {
+      if (onTitleChange) onTitleChange(e)
+    },
+    [onTitleChange]
+  )
+
+  // put the provider in readonly mode if requested and react to changes of values
+  useMemo(() => collabProvider && collabProvider.setReadOnly(!!readOnly), [
+    readOnly,
+    collabProvider
+  ])
+
+  const collabEdit = useMemo(
+    () =>
+      collabProvider && {
+        useNativePlugin: true,
+        provider: Promise.resolve(collabProvider),
+        inviteToEditHandler: () => undefined,
+        isInviteToEditButtonSelected: false,
+        userId: collabProvider.serviceClient.getSessionId()
+      },
+    [collabProvider]
+  )
+
+  useEffect(() => updateTextareaHeight(titleEl.current), [title])
+
+  useEventListener(titleEl.current, 'blur', onTitleBlur)
+
+  const provider: ImageUploadProvider = (_event, insertImageFn) => {
+    const inputElement = document.createElement('input')
+
+    inputElement.type = 'file'
+
+    inputElement.addEventListener('change', () => {
+      const reader = new window.FileReader()
+      const file = inputElement.files[0]
+
+      reader.readAsArrayBuffer(file)
+
+      reader.onloadend = async () => {
+        const server = await serviceClient.postImage(
+          file.name,
+          noteId,
+          reader.result
+        )
+
+        insertImageFn({ src: `${server.data.id}` })
+      }
+    })
+
+    inputElement.dispatchEvent(new MouseEvent('click'))
+  }
+
+  window.document.querySelectorAll('.img-wrapper img').forEach(async image => {
+    const server = await serviceClient.getImage(image.getAttribute('src'))
+    image.src = `http://cozy.localhost:8080${server.data.links.small}`
```

Comment:
We have `http://cozy.localhost:8080` hardcoded at several places. Can you fix that to make it work with other Cozy instances?

Reply:
Yes indeed, I was going to update it after verifying everything worked